components/composer/src/Composer.svelte

```diff
@@ -221,7 +221,23 @@
 
   $: if (value) {
     mergeMessage(value);
+    if (value.files?.length > 0) {
+      mergeMessage({ file_ids: value.files.map(({ id }) => id) });
+      resetAttachments();
```

Comment:
I wasn't sure if this was a good idea or not tbh. This means that if you upload a file and then have the `values` prop change somehow, you'll lose that uploaded file... But I don't think that's a big issue 🤔

Reply:
I think the proper fix here is to avoid using the store to render our uploaded attachments. We need a local variable to track this. 

The store should be used as a cache.

Reply:
Sounds good to me!

Reply:
> Without this, if you are using the same composer instance but changing the prop.value to load a new message data, you'll keep adding files to the attachments store and your composer will display attachments of the previous/old message as well.

Wouldn't the solution to this be to remove the attachments from the composer instance when the composer 
closes or the `prop.value` changes entirely?

Reply:
This does the 2nd part "or the`prop.value changes`. Right now what we do is keep adding to the store every time `prop.value` changes. So it keeps the old values in our $files store. This just resets the file store.

You are right, we could call the `close()` function to trigger the reset, but I don't think that should be expected when `prop.value` is mutated tbh. 

But perhaps a more intelligent way of checking if `prop.value.files` has changed _only_, instead of resetting it every time `prop.value.*` properties change.

Reply:
Maybe we could keep the previous attachment we uploaded if the `value` prop change occurs in the same thread. And add `composer.close();` to the `returnToMailbox` EventListener for this demo (line 41 of [mailbox/src/index.html](https://github.com/nylas/components/pull/398/files#diff-c581fcb635d197f2f77d850597322a79ed66aebfaec58a2b7e1f919e8ebde8a2)) so that it keeps the new attachment within the thread and always resets the attachment when clicking on a different thread.

Reply:
I'm trying to avoid having the user make modifications on their end to have this functionality work as expected. IMO, if I change the `value` prop, it should kinda reset the state for everything else that was previously there (e.g. to, from, subject, etc. get reset unless I choose to carry that info over).

I'm gonna leave this for now though, I can't think of a better way to implement this with the current solution.